### PR TITLE
[ext] New Getting Started tutorial (extension service worker)

### PR DESCRIPTION
**Pull request by AmySteam**
## Scope

- Adds a new tutorial to the Getting Started guides called "Quick API reference"

## TODO:

- [x] Finish Test that it works / Potential enhancements
- [x] Add links to the new explainer.
- [x] Merge code sample [PR#851](https://github.com/GoogleChrome/chrome-extensions-samples/pull/851)

### CODE REVIEW

**Review by sebastianbenz — COMMENTED** [review]
Looking good, left a few in between meetings comments. Will continue later.

File: site/en/docs/extensions/mv3/getstarted/tut-quick-reference/index.md
Comment (AmySteam) [reply site/en/docs/extensions/mv3/getstarted/tut-quick-reference/index.md:115]: Good suggestion! 👍

File: site/en/docs/extensions/mv3/getstarted/tut-quick-reference/index.md
Comment (sebastianbenz) [inline site/en/docs/extensions/mv3/getstarted/tut-quick-reference/index.md:148]:
```suggestion
The following code save the default suggestions to storage when the extension is first installed by listening to the `onInstalled` event:
```

File: site/en/docs/extensions/mv3/getstarted/tut-quick-reference/index.md
Comment (sebastianbenz) [inline site/en/docs/extensions/mv3/getstarted/tut-quick-reference/index.md:167]:
```suggestion
global variables. Instead, we use `chrome.storage.local` which stores data on the local machine.
```

File: site/en/docs/extensions/mv3/getstarted/tut-quick-reference/index.md
Comment (sebastianbenz) [inline site/en/docs/extensions/mv3/getstarted/tut-quick-reference/index.md:173]: Not sure if this is a better way to phrase this:

```suggestion
All event listeners need to be statically registered in the global scope of the service worker. In other words, event listeners should not be nested in async functions. This way Chrome can ensure that all event handlers are restored in case of a service worker reboot.
```

Maybe rephrase to:

"

File: site/en/docs/extensions/mv3/getstarted/tut-quick-reference/index.md
Comment (sebastianbenz) [inline site/en/docs/extensions/mv3/getstarted/tut-quick-reference/index.md:227]: Easier way (I think):

```
const { apiSuggestions } = await chrome.storage.local.get('apiSuggestions');
apiSuggestions = input.concat(apiSuggestions);
await chrome.storage.local.set({ apiSuggestions });
```

File: site/en/docs/extensions/mv3/getstarted/tut-quick-reference/index.md
Comment (sebastianbenz) [inline site/en/docs/extensions/mv3/getstarted/tut-quick-reference/index.md:194]: The code also creates suggestions. Maybe add a bit more explanation and maybe even split up the code.

File: site/en/docs/extensions/mv3/getstarted/tut-quick-reference/index.md
Comment (sebastianbenz) [inline site/en/docs/extensions/mv3/getstarted/tut-quick-reference/index.md:244]:
```suggestion
To use the Alarms API, request the `"alarms"` permission in the manifest. The extension also needs to request [host permission][doc-host-perm] to be able to fetch the extension tips from a remote hosted location:
```

File: site/en/docs/extensions/mv3/getstarted/tut-quick-reference/index.md
Comment (sebastianbenz) [inline site/en/docs/extensions/mv3/getstarted/tut-quick-reference/index.md:268]: remove the await here, but return the promise from set

File: site/en/docs/extensions/mv3/getstarted/tut-quick-reference/index.md
Comment (sebastianbenz) [inline site/en/docs/extensions/mv3/getstarted/tut-quick-reference/index.md:257]: Maybe mention that you create the alarm at install time

File: site/en/docs/extensions/mv3/getstarted/tut-quick-reference/index.md
Comment (sebastianbenz) [inline site/en/docs/extensions/mv3/getstarted/tut-quick-reference/index.md:292]:
```suggestion
[Content scripts][doc-content] communicate with the rest of the extension through [message passing][doc-messages]. In this example, the content script will request the tip of the day from the service worker when the user browses the Chrome extension documentation. 
```

File: site/en/docs/extensions/mv3/getstarted/tut-quick-reference/index.md
Comment (sebastianbenz) [inline site/en/docs/extensions/mv3/getstarted/tut-quick-reference/index.md:311]: Maybe mention that content scripts can't access storage.local.

File: site/en/docs/extensions/mv3/getstarted/tut-quick-reference/index.md
Comment (sebastianbenz) [inline site/en/docs/extensions/mv3/getstarted/tut-quick-reference/index.md:360]:
```suggestion
The final step is to add a message handler to our service worker that replies the daily tip when requested from the content script. 
```

**Review by sebastianbenz — COMMENTED** [review]
Few more comments, only small things. Overall I think this is great!

File: site/en/docs/extensions/mv3/getstarted/tut-quick-reference/index.md
Comment (AmySteam) [reply site/en/docs/extensions/mv3/getstarted/tut-quick-reference/index.md:227]: concat behaves funny 🙃
Screenshot 2023-03-22 at 09 19 31

File: site/en/docs/extensions/mv3/getstarted/tut-quick-reference/index.md
Comment (sebastianbenz) [reply site/en/docs/extensions/mv3/getstarted/tut-quick-reference/index.md:3]: I still think we should use a more descriptive name here. Afaik, title will be used in the article listings and it'd be great to be able to tell from the title what this tutorial. is about (please correct me if I'm wrong).

File: site/en/docs/extensions/mv3/getstarted/tut-quick-reference/index.md
Comment (sebastianbenz) [reply site/en/docs/extensions/mv3/getstarted/tut-quick-reference/index.md:228]: Sorry, should've been `[input].concat`. Thinking. more about this, I would ignore the number of suggestions and just go with:

```
const { apiSuggestions } = await chrome.storage.local.get('apiSuggestions');
await chrome.storage.local.set({ [input].concat(apiSuggestions) });
```

But your call :-)

Comment (AmySteam) [reply site/en/docs/extensions/mv3/getstarted/tut-quick-reference/index.md:228]: Okie dokies 😁

File: site/en/docs/extensions/mv3/getstarted/tut-quick-reference/index.md
Comment (AmySteam) [reply site/en/docs/extensions/mv3/getstarted/tut-quick-reference/index.md:3]: I'm making the change here and also changing the rest of the Getting Started guide tutorials to match the new approach. Thanks for your suggestion 👍

Comment (AmySteam) [reply site/en/docs/extensions/mv3/getstarted/tut-quick-reference/index.md:3]: [PR#5769](https://github.com/GoogleChrome/developer.chrome.com/pull/5769)

File: site/en/docs/extensions/mv3/getstarted/tut-quick-reference/index.md
Comment (AmySteam) [reply site/en/docs/extensions/mv3/getstarted/tut-quick-reference/index.md]: I rewrote the list 👍

File: site/en/docs/extensions/mv3/getstarted/tut-quick-reference/index.md
Comment (AmySteam) [reply site/en/docs/extensions/mv3/getstarted/tut-quick-reference/index.md:58]: Done 👍

File: site/en/docs/extensions/mv3/getstarted/tut-quick-reference/index.md
Comment (AmySteam) [reply site/en/docs/extensions/mv3/getstarted/tut-quick-reference/index.md:170]: Added this on a separate commit 👍

File: site/en/docs/extensions/mv3/getstarted/tut-quick-reference/index.md
Comment (AmySteam) [reply site/en/docs/extensions/mv3/getstarted/tut-quick-reference/index.md:197]: Good idea. On it! 🏃🏻‍♀️💨

File: site/en/docs/extensions/mv3/getstarted/tut-quick-reference/index.md
Comment (AmySteam) [reply site/en/docs/extensions/mv3/getstarted/tut-quick-reference/index.md:271]: I'm going to go through and update the code to match the sample repo. On it! 🏃🏻‍♀️💨

File: site/en/docs/extensions/mv3/getstarted/tut-quick-reference/index.md
Comment (AmySteam) [reply site/en/docs/extensions/mv3/getstarted/tut-quick-reference/index.md:336]: Technically, they DO have access to the storage API, but for security reasons, we are encouraged to use messages instead.

**Comment by AmySteam** [comment]
@sebastianbenz Thanks for sharing your insights! 💯 This is ready for another round of tech review 😄

File: site/en/docs/extensions/mv3/getstarted/tut-quick-reference/index.md
Comment (sebastianbenz) [inline site/en/docs/extensions/mv3/getstarted/tut-quick-reference/index.md:140]:
```suggestion
Chrome will shut down service workers if they are not needed. We use the [`chrome.storage`][api-storage] API to persist state across service worker sessions. For storage access, we need to request permission in the manifest:
```
I like to give a tiny bit of the why when starting a new step in a tutorial - wdyt?

File: site/en/docs/extensions/mv3/getstarted/tut-quick-reference/index.md
Comment (sebastianbenz) [inline site/en/docs/extensions/mv3/getstarted/tut-quick-reference/index.md:170]:
```suggestion
global variables.  Instead, we use [`chrome.storage.local`](/docs/extensions/reference/storage/#property-local) which stores data on the local machine.
```

File: site/en/docs/extensions/mv3/getstarted/tut-quick-reference/index.md
Comment (sebastianbenz) [inline site/en/docs/extensions/mv3/getstarted/tut-quick-reference/index.md:198]: Have we already got these links?

File: site/en/docs/extensions/mv3/getstarted/tut-quick-reference/index.md
Comment (sebastianbenz) [inline site/en/docs/extensions/mv3/getstarted/tut-quick-reference/index.md:202]:
```suggestion
Now, let's register the omnibox event listeners at the top level of the script. When the user enters the omnibox keyword `api` in the address bar followed by tab or space, Chrome will display a list of suggestions based on the keywords in storage. The [onInputChanged()][omnibox-input-changed] event, which takes the current user input and a suggestResult object, is responsible for populating these suggestions.
```

File: site/en/docs/extensions/mv3/getstarted/tut-quick-reference/index.md
Comment (sebastianbenz) [inline site/en/docs/extensions/mv3/getstarted/tut-quick-reference/index.md:265]:
```suggestion
Start by requesting the `alarms` permission in the manifest. Additionally, to fetch the extension tips from a remote hosted location, you need to request [host permission][doc-host-perm]:
```

File: site/en/docs/extensions/mv3/getstarted/tut-quick-reference/index.md
Comment (sebastianbenz) [inline site/en/docs/extensions/mv3/getstarted/tut-quick-reference/index.md:278]:
```suggestion
The extension will fetch all the tips, pick one at random and save it to storage. We will create an alarm that will be triggered once a day to update the tip. Alarms are not saved when you close Chrome. So we need to check if the alarm exists and create it if it doesn't.
```

File: site/en/docs/extensions/mv3/getstarted/tut-quick-reference/index.md
Comment (sebastianbenz) [inline site/en/docs/extensions/mv3/getstarted/tut-quick-reference/index.md:320]:
```suggestion
Extensions use [content scripts][doc-content] to read and modify the content of the page. When a user visits a Chrome API reference page, the extension's content script will update the page with the tip of the day. It [sends a message][doc-messages] to request the tip of the day from the service worker. 
```

File: site/en/docs/extensions/mv3/getstarted/tut-quick-reference/index.md
Comment (sebastianbenz) [inline site/en/docs/extensions/mv3/getstarted/tut-quick-reference/index.md:383]:
```suggestion
  if (message.type === 'tip') {
```

File: site/en/docs/extensions/mv3/getstarted/tut-quick-reference/index.md
Comment (sebastianbenz) [inline site/en/docs/extensions/mv3/getstarted/tut-quick-reference/index.md:346]:
```suggestion
  const { tip } = await chrome.runtime.sendMessage({ type: 'tip' });
```

File: site/en/docs/extensions/mv3/getstarted/tut-quick-reference/index.md
Comment (sebastianbenz) [inline site/en/docs/extensions/mv3/getstarted/tut-quick-reference/index.md:468]: Have you already got a list of articles to link to here?

**Review by sebastianbenz — COMMENTED** [review]
Few comments, but looks great! 

I've noticed that the videos are quite small. Does dcc support opening inline videos in a lightbox?

File: site/en/docs/extensions/mv3/getstarted/tut-quick-reference/index.md
Comment (AmySteam) [reply site/en/docs/extensions/mv3/getstarted/tut-quick-reference/index.md:11]: Done! 👍 Also made the video larger

File: site/en/docs/extensions/mv3/getstarted/tut-quick-reference/index.md
Comment (AmySteam) [reply site/en/docs/extensions/mv3/getstarted/tut-quick-reference/index.md:128]: Good idea... forgot about that! 😅 Thanks for pointing it out 👍

File: site/en/docs/extensions/mv3/getstarted/tut-quick-reference/index.md
Comment (AmySteam) [reply site/en/docs/extensions/mv3/getstarted/tut-quick-reference/index.md]: 👍 I added this to a fix links batch

File: site/en/docs/extensions/mv3/getstarted/tut-quick-reference/index.md
Comment (AmySteam) [reply site/en/docs/extensions/mv3/getstarted/tut-quick-reference/index.md]: Fixed 👍

File: site/en/docs/extensions/mv3/getstarted/tut-quick-reference/index.md
Comment (AmySteam) [reply site/en/docs/extensions/mv3/getstarted/tut-quick-reference/index.md:271]: This is the convention we've been following for permissions and manifest keys

File: site/en/docs/extensions/mv3/getstarted/tut-quick-reference/index.md
Comment (AmySteam) [reply site/en/docs/extensions/mv3/getstarted/tut-quick-reference/index.md:146]: Sounds good :)

File: site/en/docs/extensions/mv3/getstarted/tut-quick-reference/index.md
Comment (AmySteam) [reply site/en/docs/extensions/mv3/getstarted/tut-quick-reference/index.md:474]: I was hoping to link to some Service worker "How-to's" or the explainer, but that is only one approach. 

We could also simply link our to general development articles. I'm open to suggestions.

File: site/en/docs/extensions/mv3/getstarted/tut-quick-reference/index.md
Comment (AmySteam) [reply site/en/docs/extensions/mv3/getstarted/tut-quick-reference/index.md:352]: Thanks for this suggestion.  Since we already merged the sample code, I'd rather leave this as is for now. If we find a critical change to the source code, I will add this to the list.

File: site/en/docs/extensions/mv3/getstarted/tut-quick-reference/index.md
Comment (rachelandrew) [reply site/en/docs/extensions/mv3/getstarted/tut-quick-reference/index.md:115]: We don't use emphasis like this typically.

**Review by rachelandrew — APPROVED** [review]
LGTM with one small comment, I'm assuming all the TBD links will be fixed before merging.

**Pull request merged**
